pyscraper/new_hansard.py

```diff
@@ -0,0 +1,1417 @@
+        tag.set(
+            'url',
+            'http://www.publications.parliament.uk{0}'.format(
+                heading.get('url')
+            )
+        )
+        tag.text = heading.text
+        self.root.append(tag)
+
+    def parse_major(self, heading, **kwargs):
+        self.check_for_pi(heading)
+        text = self.get_text_from_element(heading)
+        if text.strip() == 'Prayers':
+            return
+        self.clear_current_speech()
+        tag = etree.Element('major-heading')
+        if self.uc_titles:
+            tag.text = text.upper()
+        else:
+            tag.text = text
+
+        if 'extra_text' in kwargs:
+            tag.text = u'{0} - '.format(tag.text)
+            i = etree.Element('i')
+            i.text = kwargs['extra_text']
+            tag.append(i)
+
+        tag.set('id', self.get_speech_id())
+        tag.set('nospeaker', 'true')
+        tag.set('colnum', self.current_col)
+        tag.set('time', self.current_time)
+        tag.set(
+            'url',
+            'http://www.publications.parliament.uk{0}'.format(
+                heading.get('url')
+            )
+        )
+        self.root.append(tag)
+        self.output_heading = True
+        self.current_speech_num = self.current_speech_num + 1
+
+    def parse_minor(self, heading):
+        self.check_for_pi(heading)
+        if self.current_speech is not None:
+            self.clear_current_speech()
+        tag = etree.Element('minor-heading')
+        tag.set('id', self.get_speech_id())
+        tag.set('nospeaker', 'true')
+        tag.set('colnum', self.current_col)
+        tag.set('time', self.current_time)
+        tag.set(
+            'url',
+            'http://www.publications.parliament.uk{0}'.format(
+                heading.get('url')
+            )
+        )
+        text = self.get_text_from_element(heading)
+        tag.text = text
+        self.root.append(tag)
+        self.output_heading = True
+        self.current_speech_num = self.current_speech_num + 1
+
+    def parse_generic(self, heading):
+        self.check_for_pi(heading)
+        if self.current_speech_num <= 0:
+            self.parse_major(heading)
+        else:
+            self.parse_minor(heading)
+
+    def parse_opposition(self, heading):
+        following = heading.xpath(
+            '(./following-sibling::ns:hs_2cDebatedMotion|./following-sibling::ns:hs_7SmCapsHdg|./following-sibling::ns:hs_2GenericHdg)',
+            namespaces=self.ns_map
+        )
+        text = ''
+        if len(following) == 1:
+            text = u' - '.join([
+                self.get_single_line_text_from_element(heading),
+                self.get_single_line_text_from_element(following[0])
+            ])
+            heading.text = text
+            self.skip_tag = self.get_tag_name_no_ns(following[0])
+
+        self.parse_major(heading)
+
+    def parse_debated_motion(self, motion):
+        following = motion.xpath(
```

Comment:
I think it would be helpful to have little excerpts of an XML example here (and other places like this), as to what it is looking for and spotting (and separately, that it's spotting everything it should).